svelte/fr/docs/05-compile-time.md

```diff
@@ -0,0 +1,367 @@
+		frame: string,
+		toString: () => string
+	}>,
+	vars: Array<{
+		name: string,
+		export_name: string,
+		injected: boolean,
+		module: boolean,
+		mutated: boolean,
+		reassigned: boolean,
+		referenced: boolean,
+		referenced_from_script: boolean,
+		writable: boolean
+	}>,
+	stats: {
+		timings: { [label]: number }
+	}
+} = svelte.compile(source: string, options?: {...})
+```
+
+-->
+
+
+### `svelte.parse`
+
+```js
+ast: object = svelte.parse(
+	source: string,
+	options?: {
+		filename?: string,
+		customElement?: boolean
+	}
+)
+```
+
+---
+
+La méthode `parse` convertit un composant pour retourner son arbre syntaxique abstrait (*abstract syntax tree*, ou *AST*). Contrairement à la compilation avec l'option `generate: false`, aucune validation ni analyse n'est appliqué. Notez que l'*AST* n'est pas considéré comme une API publique ; des changements critiques pourraient survenir à n'importe quel moment.
+
+
+```js
+const svelte = require('svelte/compiler');
+
+const ast = svelte.parse(source, { filename: 'App.svelte' });
+```
+
+
+### `svelte.preprocess`
+
+Un certain nombre de [plugins de pré-compilation maintenus par la communauté](https://sveltesociety.dev/tools#preprocessors) est dispnible pour vous permettre d'utiliser Svelte avec des outils comme TypeScript, PostCSS, SCSS, et Less.
+
+Vous pouvez écrire votre propre *préprocessor* en utilisant l'API `svelte.preprocess`.
+
+```js
+result: {
+	code: string,
+	dependencies: Array<string>
+} = await svelte.preprocess(
+	source: string,
+	preprocessors: Array<{
+		markup?: (input: { content: string, filename: string }) => Promise<{
+			code: string,
+			dependencies?: Array<string>
+		}>,
+		script?: (input: { content: string, markup: string, attributes: Record<string, string>, filename: string }) => Promise<{
+			code: string,
+			dependencies?: Array<string>
+		}>,
+		style?: (input: { content: string, markup: string, attributes: Record<string, string>, filename: string }) => Promise<{
+			code: string,
+			dependencies?: Array<string>
+		}>
+	}>,
+	options?: {
+		filename?: string
+	}
+)
+```
+
+---
+
+Le fonction de préprocessing fournit des hooks pour transformer le code source d'un composant, pour plusieurs langages. Par exemple, elle peut convertir un bloc `<style lang="sass">` en css natif.
+
+Le premier argument est le code source du composant lui-même. Le second argument est un tableau de *préprocesseurs* (ou éventuellement un seul préprocesseur si vous n'en avez qu'un). Un *préprocesseur* est un objet contenant trois fonctions optionnelles : `markup`, `script` et `style`.
+
+Chaque fonction `markup`, `script` et `style` doit retourner un objet (ou une promesse qui résout un objet) contenant un attribut `code`, représentant le code source transofrmé ainsi qu'un tableau facultatif de dépendances `dependencies`.
+
+La fonction `markup` reçoit en argument le *markup* du composant, et le nom du composant `filename` s'il était spécifié comme troisième argument.
```

Comment:
On est sûrs que c'est pas l'intégralité du code source ici ? 
Le nom de la fonction suggère que c'est uniquement le markup, mais la description "the entire component source text" me met le doute.